pkg/api/types.go

```diff
@@ -485,10 +485,17 @@ type PipelineImageCacheStepConfiguration struct {
 type TestStepConfiguration struct {
 	// As is the name of the test.
 	As string `json:"as"`
+
 	// Commands are the shell commands to run in
 	// the repository root to execute tests.
 	Commands string `json:"commands,omitempty"`
 
+	// SkipRehearsal indicates that the "can-be-rehearsed"
+	// label is to be omitted. This only applies to presubmit
+	// jobs. Postsubmit jobs will never be rehearsed. Periodic
+	// jobs will always be rehearsed.
+	SkipRehearsal bool `json:"skip_rehearsal,omitempty"`
```

Comment:
How about alternatively change your jobs so they can be rehearsed?

Reply:
The current default behavior is to always rehearse presubmit jobs. If I negated the logic here, none of the existing presubmit jobs would be rehearsed and all test configurations across all orgs and repos would need to be reviewed and updated to enable rehearsing jobs.

I can do this as it's not truly a "breaking" change, but I felt it would be better to incrementally change the behavior rather than completely upend it.

Just let me know.

Reply:
Sorry. I misread your comment earlier. I understand it now.

The problem with rehearsing these jobs is that they require $REPO_OWNER, $REPO_NAME, and $PULL_NUMBER to be set correctly to work. But when the job is rehearsed, these are set to `openshift`, `release`, and whatever the PR number is where the rehearsal is taking place.

Our jobs that run the SonarCloud scan fail because the $PULL_NUMBER in the rehearsal isn't in the actual component repo and so it fails with this type of error:
```
DEBUG: GET 404 https://sonarcloud.io/api/alm_integration/show_pullrequest?project=open-cluster-management_console_api&pullrequestKey=13425
```
where `13425` is the PR number for `openshift/release` that doesn't exist in the `open-cluster-management/console-api` repo. This isn't the primary concern though as this doesn't really "cost" us anything, but the rehearsal will never pass.

Our e2e tests are going to be the ones that cause the most problems (and cost). Here's the entry for the workflow:
https://steps.ci.openshift.org/workflow/ocm-e2e-clusterpool

I know the docs say that using these Prow variables "breaks encapsulation" but our e2e workflow (in a pre step command script) needs to know which component is being tested. It has to checkout an OCP cluster, partially deploy ACM, patch the deployment CSV for ACM with the image ref for that component, and finish the deployment. During a pre step, the only place the component info under test seems to be available is in $REPO_OWNER and $REPO_NAME.